### PR TITLE
chore(rpc): decouple op receipt response

### PR DESCRIPTION
**Pull request by emhane**
- Adds builder for RPC receipt response
- Moves optimism response body extension logic to optimism crate
- Removes deps `dep:reth-evm-optimism` and `reth-evm-optimism/optimism` from `optimism` feature in `reth-rpc`

### CODE REVIEW

**Comment by emhane** [comment]
blocked by `matt/scaffold-ethapi` which should be base, but isn't compiling yet, will change base when I got that code with skeleton code for type `OpEthApi` compiling

File: crates/optimism/rpc/src/lib.rs
Comment (mattsse) [inline crates/optimism/rpc/src/lib.rs:1]: this needs the standard header

File: crates/optimism/rpc/src/receipts.rs [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+/// Optimism Transaction Metadata
+///
+/// Includes the L1 fee and data gas for the tx along with the L1
+/// block info. In order to pass the [OptimismTxMeta] into the
+/// async colored [ReceiptResponseBuilder], a reference counter
+/// for the L1 block info is used so the L1 block info can be
+/// shared between receipts.
+#[derive(Debug, Default, Clone)]
+pub struct OptimismTxMeta {
```
Comment (mattsse) [inline crates/optimism/rpc/src/receipts.rs:174]: let's move this to transaction.rs

**Review by mattsse — DISMISSED** [review]
I like this a lot as an initial step towards rpc abstraction

have a few nits

File: crates/optimism/rpc/src/lib.rs
Comment (emhane) [reply crates/optimism/rpc/src/lib.rs:1]: not sure what you mean...I made it same as for the other optimism crates?

Comment (mattsse) [reply crates/optimism/rpc/src/lib.rs:1]: https://github.com/paradigmxyz/reth/blob/72f2f1b850a9454789f3ebb3f6efccaee4125659/crates/optimism/node/src/lib.rs#L1-L10

**Pull request merged**